### PR TITLE
Add maxRestartTimes for container

### PR DESCRIPTION
**Pull request by choury**
**What this PR does / why we need it**:
Add restart limit for container if it always crash.

**Special notes for your reviewer**:

**Release note**:

```release-note
Add restart limit for container if it always crash.
```

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
@choury is `maxRestartTimes` documented anywhere?

**Comment by choury** [comment]
@xiangpengzhao There is no document now, I will add it  if this proposal are acceptable.

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 1]
```diff
@@ -730,6 +731,10 @@ func (m *kubeGenericRuntimeManager) doBackOff(pod *v1.Pod, container *v1.Contain
 		err := fmt.Errorf("Back-off %s restarting failed container=%s pod=%s", backOff.Get(key), container.Name, format.Pod(pod))
 		glog.Infof("%s", err.Error())
 		return true, err.Error(), kubecontainer.ErrCrashLoopBackOff
+		maxRestartTimes, err := strconv.Atoi(pod.ObjectMeta.Annotations["maxRestartTimes"])
```
Comment (xiangpengzhao) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go]: The code you added will never be exec'd because there is a `return` ahead of it...

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go (hunk 1 above)
Comment (choury) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go]: Oh, That's a merge mistake, I will fix it as soon as possible.

Comment (choury) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go]: Sorry for such stupid mistake, updated now.

Comment (xiangpengzhao) [reply pkg/kubelet/kuberuntime/kuberuntime_manager.go]: :)

**Pull request closed without merging**